src/e_ibmca.c

```diff
@@ -42,6 +42,15 @@
  #define OLDER_OPENSSL
 #endif
 
+/*
+ * Just add DEBUG_PRINTF statements to the code and comment the
+ * 1st define and uncomment the 2nd define and you have a very
+ * simple debugging possibility. Don't forget to switch off before
+ * pushing your great code upstream.
+*/
+#define DEBUG_PRINTF(...) do{} while(0)
+//#define DEBUG_PRINTF(...) printf(__VA_ARGS__)
+
```

Comment:
Instead of doing this manually, I would prefer a debug mechanism that is enabled when you compile the project: 
./configure --enable-debug 

openssl-ibmpkcs11 has this, and I believe ibmca should follow the same. 
On configure.ac you will see that we already have a macro defined DEBUG ... doing a #ifdef DEBUG should work.

Reply:
ok, I'll check this and maybe rework...

Reply:
Done. Now with ./configure --enable-debug the DEBUG_PRINTF macro is active. Without this configure option the macro simple expands to an empty do{}while(0) loop which is removed by the compiler. Tested, works.